src/raftstore/store/fsm/apply.rs

```diff
@@ -2593,18 +2566,20 @@ impl ApplyFsm {
         );
 
         let region_id = self.delegate.region_id();
-        self.destroy(ctx);
-        catch_up_logs
-            .logs_up_to_date
-            .store(region_id, Ordering::SeqCst);
         info!(
             "source logs are all applied now";
             "region_id" => region_id,
             "peer_id" => self.delegate.id(),
         );
+        // The source peerfsm will be destroyed when the target peer executes `on_ready_commit_merge` 
+        // and sends `merge result` to the source peerfsm.
+        self.destroy(ctx);
+        catch_up_logs
+            .logs_up_to_date
+            .store(region_id, Ordering::Release);
```

Comment:
Use `SeqCst` instead.

Reply:
The sequentially-consistent order is unnecessary. This is a typical scenario using acquire and release semantics.

Reply:
In TiKV, we prefer to use `SeqCst` in all places unless it's performance critical. The policy is to save mental analyze.